File: 0852_Peak Index in a Mountain Array/PeakIndexinaMountainArray.cs
public class Solution {
    //Find the smallest l such that arr[l] > arr[l + 1].

    public int PeakIndexInMountainArray(int[] arr) {
        var l = 0;
        var r = arr.Length;
        while(l < r){
            var m = (r-l)/2 + l;
            if(arr[m] > arr[m+1]){
                r = m;
            }else{
                l = m + 1;
            }
        }
        
        return l;
    }
}

File: 0852_Peak Index in a Mountain Array/PeakIndexinaMountainArray.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int l = 0;
        int r = arr.length - 1;
        
        while(l < r)
        {
            int mid = (r-l)/2 + l;
            if(arr[mid] > arr[mid+1])
                r = mid;
            else
                l = mid+1;
        }
        
        return l;
    }
}